### PR TITLE
[theoros] feat: Hyperlane types

### PR DESCRIPTION
**Pull request by akhercha**
Resolves: #NA

## Work done

* Added needed structures for the Hyperlane validators checkpoints etc

### CODE REVIEW

File: rust/pragma-utils/src/conversions/starknet.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
+use anyhow::{bail, Result};
```
Comment (EvolveArt) [inline rust/pragma-utils/src/conversions/starknet.rs:1]: too complex implementation let's leverage methods within the Felt crate or the macro `short_string`

Comment (EvolveArt) [reply rust/pragma-utils/src/conversions/starknet.rs:1]: lets create an issue for now and revisit later

File: rust/theoros/src/types/starknet_rpc.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    macros::selector,
+    providers::{jsonrpc::HttpTransport, JsonRpcClient, Provider},
+};
+use url::Url;
+
+use super::StorageLocation;
+
+pub struct StarknetRpc(JsonRpcClient<HttpTransport>);
+
+impl StarknetRpc {
```
Comment (EvolveArt) [inline rust/theoros/src/types/starknet_rpc.rs:14]: let's create an issue for using cainome later on, for safer interaction with contracts

**Review by EvolveArt — APPROVED** [review]
LGTM just 2 issues to create

File: rust/theoros/src/types/starknet_rpc.rs (hunk 2 above)
Comment (akhercha) [reply rust/theoros/src/types/starknet_rpc.rs:14]: added here:
https://github.com/astraly-labs/pragma-monorepo/issues/22

File: rust/pragma-utils/src/conversions/starknet.rs (hunk 1 above)
Comment (akhercha) [reply rust/pragma-utils/src/conversions/starknet.rs:1]: added here:
https://github.com/astraly-labs/pragma-monorepo/issues/23

**Pull request merged**